packages/dd-trace/test/plugins/agent.js

```diff
@@ -234,36 +234,37 @@ let availableEndpoints = DEFAULT_AVAILABLE_ENDPOINTS
  * @param {Set} [handlers] - Set of handlers to add the callback to.
  * @returns {Promise<void>} A promise resolving if expectations are met
  */
-function runCallback (callback, options, handlers) {
-  const deferred = {}
-  const promise = new Promise((resolve, reject) => {
-    deferred.resolve = resolve
-    deferred.reject = reject
-  })
+function runCallbackAgainstTraces (callback, options, handlers) {
+  let error
 
-  const timeoutMs = options !== null && typeof options === 'object' && options.timeoutMs ? options.timeoutMs : 1000
+  let resolve
+  let reject
+  const promise = new Promise((_resolve, _reject) => {
+    resolve = _resolve
+    reject = _reject
+  })
 
-  const timeout = setTimeout(() => {
-    if (error) {
-      deferred.reject(error)
-    }
-  }, timeoutMs)
+  const rejectionTimeout = setTimeout(() => {
+    if (error) reject(error)
```

Comment:
I believe error is always truthy (if not throwing something falsy). Otherwise the timeout is cleared.

```suggestion
    reject(error)
```

Reply:
Ok, thanks :)